langchain/document_loaders/weather.py

```diff
@@ -0,0 +1,50 @@
+"""Simple reader that reads weather data from OpenWeatherMap API"""
+from __future__ import annotations
+
+from datetime import datetime
+from typing import Iterable, Iterator, List, Optional
+
+from langchain.docstore.document import Document
+from langchain.document_loaders.base import BaseLoader
+from langchain.utilities.openweathermap import OpenWeatherMapAPIWrapper
+
+
+class WeatherDataLoader(BaseLoader):
+    """Weather Reader.
+
+    Reads the forecast & current weather of any location using OpenWeatherMap's free
+    API. Checkout 'https://openweathermap.org/appid' for more on how to generate a free
+    OpenWeatherMap API.
+    """
+
+    def __init__(
+        self,
+        client: OpenWeatherMapAPIWrapper,
+        places: Iterable[str],
```

Comment:
Might make sense to use a Sequence here -- i think it's reasonable to expect that we're dealing with a finite number of places that's loaded into memory